src/never.rs

```diff
@@ -0,0 +1,91 @@
+use {Future, Poll};
+
+/// A future which is never resolved.
+///
+/// This future can be created with the `empty` function.
+#[derive(Copy, Clone)]
+pub struct Never {}
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// The returned future will never resolve with a success but is still
+/// susceptible to cancellation. That is, if a callback is scheduled on the
+/// returned future, it is only run once the future is dropped (canceled).
+pub fn never<T, E>() -> impl Future<Item = T, Error = E> + Send + 'static {
```

Comment:
Hm so to clarify, I'm not really sure where this PR is going at this point, nor why we'd want to add features for the `!` type? What's the goal that this is trying to achieve at this poitn?